x/observer/types/message_update_observer.go

```diff
@@ -0,0 +1,60 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const TypeMsgUpdateObserver = "update_observer"
+
+var _ sdk.Msg = &MsgUpdateObserver{}
+
+func NewMsgUpdateObserver(creator string, oldObserverAddress string, newObserverAddress string, updateReason ObserverUpdateReason) *MsgUpdateObserver {
+	return &MsgUpdateObserver{
+		Creator:            creator,
+		OldObserverAddress: oldObserverAddress,
+		NewObserverAddress: newObserverAddress,
+		UpdateReason:       updateReason,
+	}
+}
+
+func (msg *MsgUpdateObserver) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgUpdateObserver) Type() string {
+	return TypeMsgUpdateObserver
+}
+
+func (msg *MsgUpdateObserver) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgUpdateObserver) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgUpdateObserver) ValidateBasic() error {
```

Comment:
Maybe we can also returns an error if reason is different than admin update and tombstoned?